param/__init__.py

```diff
@@ -2378,6 +2378,25 @@ def __init__(self, default=Undefined, bounds=Undefined, softbounds=Undefined,
     def _validate(self, val):
         super(Range, self)._validate(val)
         self._validate_bounds(val, self.bounds, self.inclusive_bounds)
+        self._validate_step(val, self.step)
+        self._validate_order(val, allow_None=self.allow_None)
+
+    def _validate_step(self, val, step):
+        if step is not None and not _is_number(step):
+            raise ValueError("Step can only be None or a "
+                             "numeric value, not type %r." % type(step))
+
+    def _validate_order(self, val, allow_None):
+        if val is None and allow_None:
+            return
+        elif val is not None and (val[0] is None or val[1] is None):
+            return
+
+        start, end = val
+        if not start <= end:
```

Comment:
Actually, @jlstevens, I'm having a bit of a doubt here. Should `Range` enforce `a <= b` or is it actually fine to have `a < b`? Like in Python `range(10, 0, -1)` which would "translate" in Param to `param.Range(default=(10, 0), step=-1)`?

The code in the existing `_validate_bounds` seems to assume `a <= b`: https://github.com/holoviz/param/pull/725/files#diff-b4344305e71345c3583ee7c047c524b6b272f31f6af55e2b14c54914b1476cd1R2406-R2408

Reply:
What seems natural is to enforce a<=b if step>0, enforce b=>a if step<0, and enforce b=a if step=0.

Reply:
Yes that makes sense. Even if the case where step=0 is a little weird. Python `range` does not accept a step of 0.

Reply:
Disallowing step=0 altogether also works; as you wish.

Reply:
I think the semantics Jim suggested are good and I would vote for disabling `step=0` entirely.

Reply:
Thanks for the comments 👍 Latest commit should implement what is described above.